src/tactic/core.lean

```diff
@@ -706,25 +706,42 @@ meta def iterate_at_most_on_subgoals : nat → tactic unit → tactic unit
 | 0        tac := trace "maximal iterations reached"
 | (succ n) tac := focus1 (do tac, iterate_at_most_on_all_goals n tac)
 
-/-- `apply_list l`: try to apply the tactics in the list `l` on the first goal, and
-fail if none succeeds -/
-meta def apply_list_expr : list expr → tactic unit
+/-- This makes sure that the execution of the tactic does not change the tactic state.
+This can be helpful while using rewrite, apply, or expr munging.
+Remember to instantiate your metavariables before you're done! -/
+meta def lock_tactic_state {α} (t : tactic α) : tactic α
+| s := match t s with
+       | result.success a s' := result.success a s
+       | result.exception msg pos s' := result.exception msg pos s
+end
+
+/--
+`apply_list l`, for `l : list (tactic expr)`,
+tries to apply the lemmas generated by the tactics in `l` on the first goal, and
+fail if none succeeds.
+-/
+meta def apply_list_expr : list (tactic expr) → tactic unit
```

Comment:
Given that you've changed the type, shouldn't we also change the name (both here and in the docstring -- which by the way already doesn't have the right name)?

Reply:
Changed both!